python/tskit/metadata.py

```diff
@@ -516,14 +516,20 @@ def __init__(self, schema: Optional[Mapping[str, Any]]) -> None:
             # does not.
             schema = codec_cls.modify_schema(schema)
             codec_instance = codec_cls(schema)
-            self._string = json.dumps(schema)
+            self._string = json.dumps(schema, sort_keys=True)
             self._validate_row = TSKITMetadataSchemaValidator(schema).validate
             self.encode_row = codec_instance.encode
             self.decode_row = codec_instance.decode
 
     def __str__(self) -> str:
         return self._string
 
+    def __eq__(self, other) -> bool:
+        return self._string == other._string
+
+    def __ne__(self, other) -> bool:
```

Comment:
I don't think there's any need to implement ``__ne__`` as well since Python 3: https://docs.python.org/3/reference/datamodel.html#object.__ne__

Reply:
Nice, removed.